File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        totalSum = sum(nums)
        prefixSum = 0
        for i in range(len(nums)):
            if prefixSum == totalSum - prefixSum - nums[i]:
                return i
            prefixSum += nums[i]
        return -1        

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int pivot=-1;
        int[] arr=new int[nums.length];
        int pSum=0;
        for(int j=0;j<nums.length;j++){
            arr[j]=pSum;
            pSum+=nums[j];
        }
        //pSum+=nums[nums.length-1];
        for(int j=0;j<nums.length ;j++){
            if(pSum-nums[j]-arr[j]==arr[j]) return j;

        }
        return -1;
    }
}